kusto/ingest/managed.go

```diff
@@ -44,11 +46,72 @@ func NewManaged(client QueryClient, db, table string, options ...Option) (*Manag
 	}, nil
 }
 
+// Attempts to stream with retries, on success - return res,nil.
+// If failed permanently - return err,nil.
+// If failed transiently - return nil,nil.
+func (m *Managed) streamWithRetries(ctx context.Context, payloadProvider func() io.Reader, props properties.All, isBlobUri bool) (*Result, error) {
+	var result *Result
+
+	hasCustomId := props.Streaming.ClientRequestId != ""
+	i := 0
+	managedUuid := uuid.New().String()
+
+	actualBackoff := backoff.WithContext(backoff.WithMaxRetries(props.ManagedStreaming.Backoff, retryCount), ctx)
+
+	var err error = nil
+	err = backoff.Retry(func() error {
+		if !hasCustomId {
+			props.Streaming.ClientRequestId = fmt.Sprintf("KGC.executeManagedStreamingIngest;%s;%d", managedUuid, i)
+		}
+		result, err = streamImpl(m.streaming.streamConn, ctx, payloadProvider(), props, isBlobUri)
+		i++
+		if err != nil {
+			if e, ok := err.(*errors.Error); ok {
+				if errors.Retry(e) {
+					return err
+				} else {
+					return backoff.Permanent(err)
+				}
+			} else {
+				return backoff.Permanent(err)
+			}
+		}
+		return nil
+	}, actualBackoff)
+
+	if err == nil {
+		return result, nil
+	}
+
+	if errors.Retry(err) {
+		// Caller should fallback to queued
+		return nil, nil
+	}
+
+	return nil, err
+}
+
 func (m *Managed) FromFile(ctx context.Context, fPath string, options ...FileOption) (*Result, error) {
 	props := m.newProp()
 	file, err := prepFileAndProps(fPath, &props, options, ManagedClient)
 
-	if err == FileIsBlobErr { // Non-local file - fallback to queued
+	if file == nil {
+		if props.Ingestion.RawDataSize == 0 {
+			size, err := resources.FetchBlobSize(fPath, ctx, m.queued.client.HttpClient())
+			if err != nil {
+				// Failed fetch blob properties
+				return nil, err
+			}
+			props.Ingestion.RawDataSize = size
+		}
+
+		if props.Ingestion.RawDataSize <= maxStreamingSize {
```

Comment:
You are not taking into account compression.

Reply:
as said above - this needs to be done in another pr